docs/user-guide/configuring/configuration-files.md

```diff
@@ -0,0 +1,393 @@
+
+The second way to use configuration files is to save the file wherever you would like and pass its location to the CLI using the `-c` option, such as:
+
+    eslint -c myconfig.json myfiletotest.js
+
+If you are using one configuration file and want ESLint to ignore any `.eslintrc.*` files, make sure to use `--no-eslintrc` along with the `-c` flag.
+
+In each case, the settings in the configuration file override default settings.
+
+### Comments in configuration files
+
+Both the JSON and YAML configuration file formats support comments (`package.json` files should not include them). You can use JavaScript-style comments or YAML-style comments in either type of file and ESLint will safely ignore them. This allows your configuration files to be more human-friendly. For example:
+
+```js
+{
+    "env": {
+        "browser": true
+    },
+    "rules": {
+        // Override our default settings just for this directory
+        "eqeqeq": "warn",
+        "strict": "off"
+    }
+}
+```
+
+## Adding Shared Settings
+
+ESLint supports adding shared settings into configuration files. You can add `settings` object to ESLint configuration file and it will be supplied to every rule being executed. This may be useful if you are adding custom rules and want them to have access to the same information and be easily configurable.
+
+In JSON:
+
+```json
+{
+    "settings": {
+        "sharedData": "Hello"
+    }
+}
+```
+
+And in YAML:
+
+```yaml
+---
+  settings:
+    sharedData: "Hello"
+```
+
+## Cascading and Hierarchy
+
+When using `.eslintrc.*` and `package.json` files for configuration, you can take advantage of configuration cascading. Suppose you have the following structure:
+
+```text
+your-project
+├── .eslintrc.json
+├── lib
+│ └── source.js
+└─┬ tests
+  ├── .eslintrc.json
+  └── test.js
+```
+
+The configuration cascade works by using the closest `.eslintrc` file to the file being linted as the highest priority, then any configuration files in the parent directory, and so on. When you run ESLint on this project, all files in `lib/` will use the `.eslintrc` file at the root of the project as their configuration. When ESLint traverses into the `tests/` directory, it will then use `your-project/tests/.eslintrc` in addition to `your-project/.eslintrc`. So `your-project/tests/test.js` is linted based on the combination of the two `.eslintrc` files in its directory hierarchy, with the closest one taking priority. In this way, you can have project-level ESLint settings and also have directory-specific overrides.
```

Comment:
Does something like the following work?

> The configuration cascade works by using the project-wide `eslintrc` file to the file being linted then any configuration files in the parent directory, and so on.

Reply:
I don't think that suggestion is specific enough, but we're on the right track. The following is too wordy, but maybe someone else can find a way to pare it down for me?

> ESLint allows for configuration files in subdirectories to override values in the project-wide configuration when linting files in that subdirectory. Please note that only the configuration options defined in the subdirectories will be overridden. More specifically, ESLint starts from the  root configuration (the configuration file with `root: true` defined, usually in the root directory of the project), and if it finds a configuration file in a subdirectory, it will merge those values into the root configuration when linting the files in that subdirectory.

Reply:
@nzakas This is the only issue that needs to be resolved.

Reply:
Yeah, I think we can make this a bit clearer. Here's my take:

> The configuration cascade works based on the location of the file being linted. If there is a `.eslintrc` file in the same directory as the file being linted, then that configuration takes precedence. ESLint then searches up the directory structure, merging any `.eslintrc` files it finds along the way until reaching either a `.eslintrc` file with `root: true` or the root directory.